### PR TITLE
Vdh Feed the Demon and Gluttony Talents

### PR DESCRIPTION
**Pull request by yajinni**
Feed the Demon log:
https://www.warcraftlogs.com/reports/ZVJr2MPNx3RCvX6B/#fight=6&source=184
image

Gluttony Log:
https://www.warcraftlogs.com/reports/rz6WxLbAmTgnFXQP/#fight=3&source=3
image

### CODE REVIEW

File: src/Parser/DemonHunter/Vengeance/Modules/Features/Checklist/Component.js [hunk 1]
```diff
@@ -42,37 +44,102 @@ class VengeanceDemonHunterChecklist extends React.PureComponent {
...
-          <AbilityRequirement spell={SPELLS.SIGIL_OF_FLAME_CONCENTRATED.id} />
-        </Rule>
+          {combatant.hasTalent(SPELLS.SPIRIT_BOMB_TALENT.id) && !combatant.hasTalent(SPELLS.FEED_THE_DEMON_TALENT.id) &&(
+            <Requirement
+              name={(
+                <React.Fragment>
+                  <SpellLink id={SPELLS.SPIRIT_BOMB_TALENT.id} /> 4+ souls casts
+                </React.Fragment>
+              )}
+              thresholds={thresholds.spiritBombSoulConsume}
```
Comment (jjs451) [inline src/Parser/DemonHunter/Vengeance/Modules/Features/Checklist/Component.js]: Possible typo, you might have meant souls

Comment (yajinni) [reply src/Parser/DemonHunter/Vengeance/Modules/Features/Checklist/Component.js]: Technically EVERYTHING is using that. So i made a typo in its original creation ahaha

File: src/Parser/DemonHunter/Vengeance/Modules/Features/SoulFragmentsTracker.js [hunk 2]
```diff
@@ -34,7 +40,28 @@ class SoulFragmentsTracker extends Analyzer {
       this.soulsSpent += 1;
       this.currentSouls -= 1;
     }
+    if (!this.selectedCombatant.hasTalent(SPELLS.FEED_THE_DEMON_TALENT.id)){
+      return;
+    }
+    if (!this.spellUsable.isOnCooldown(SPELLS.DEMON_SPIKES.id)){
+      this.totalCooldownReductionWasted += COOLDOWN_REDUCTION_MS;
+      console.log("TEST ", this.totalCooldownReductionWasted);
```
Comment (jjs451) [inline src/Parser/DemonHunter/Vengeance/Modules/Features/SoulFragmentsTracker.js]: This should be removed or disabled if you need to save it for future debugging

File: src/Parser/DemonHunter/Vengeance/Modules/Talents/FeedTheDemon.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import { formatPercentage, formatNumber } from 'common/format';
+import AbilityTracker from 'Parser/Core/Modules/AbilityTracker';
+
+import StatisticBox, { STATISTIC_ORDER } from 'Interface/Others/StatisticBox';
+import SoulFragmentsTracker from '../Features/SoulFragmentsTracker';
+
+//WCL https://www.warcraftlogs.com/reports/ZVJr2MPNx3RCvX6B/#fight=6&source=184
+class FeedTheDemon extends Analyzer {
+  static dependencies = {
+    soulFragmentsTracker: SoulFragmentsTracker,
```
Comment (jjs451) [inline src/Parser/DemonHunter/Vengeance/Modules/Talents/FeedTheDemon.js]: I would recommend removing the soulFragmentsTracker dependency and moving all your FTD code into one file.  As best I can tell, the tracker isn't needed to do the FTD analysis but is just a convenient block already setup to fire when soul fragment stacks are removed.  To keep things modular and neat, you should take what you added in the soul fragment tracker and move it in here.  It looks good to go so it should be a quick copy/paste and commit.

**Comment by yajinni** [comment]
made the requested changes

**Comment by yajinni** [comment]
updated statistic order

**Pull request merged**